### ISSUE TITLE
ZeroDivisionError: division by zero

### ISSUE THREAD

**Issue by WenFuLee** [uber-research/UPSNet#22]
When doing an evaluation on the test set, I got the following error. I don't have any clue about it.

```
2019-04-18 03:56:17,288 | upsnet_end2end_test.py | line 307: unified pano result:
Traceback (most recent call last):
  File "upsnet/upsnet_end2end_test.py", line 316, in <module>
    upsnet_test()
  File "upsnet/upsnet_end2end_test.py", line 308, in upsnet_test
    test_dataset.evaluate_panoptic(test_dataset.get_unified_pan_result(all_ssegs, all_panos, all_pano_cls_inds, stuff_area_limit=config.test.panoptic_stuff_area_limit), os.path.join(final_output_path, 'results', 'pans_unified'))
  File "upsnet/../upsnet/dataset/base_dataset.py", line 333, in evaluate_panoptic
    results = pq_compute(gt_json, pred_json, gt_pans, pred_pans, categories)
  File "upsnet/../upsnet/dataset/base_dataset.py", line 301, in pq_compute
    results[name], per_class_results = pq_stat.pq_average(categories, isthing=isthing)
  File "upsnet/../upsnet/dataset/base_dataset.py", line 97, in pq_average
    return {'pq': pq / n, 'sq': sq / n, 'rq': rq / n, 'n': n}, per_class_results
ZeroDivisionError: division by zero
```
But, the panoptic segmentation results can be successfully generated like below.
![lindau_000000_000019](https://user-images.githubusercontent.com/34199594/56339661-c7efe880-6174-11e9-9088-16c6b5caff0f.png)

So, how is that possible that I run into the case n = 0? Any idea would be appreciated...Thanks.

**Comment by YuwenXiong** [comment]
please check https://github.com/uber-research/UPSNet/issues/11#issuecomment-481955737

**Comment by WenFuLee** [comment]
I followed #11 (comment), but the solutions in the comments didn't work out for me.

**Comment by WenFuLee** [comment]
This issue was solved according to #11 .

**Comment by discretecoder** [comment]
I am also having this error with my own dataset. 
how can I resolve the issue

**Comment by rlangefe** [comment]
I ran into this issue too when trying to use my own dataset. Did anyone find a solution to this?

**Comment by RYYAI** [comment]
Same problem here as well.

**Comment by rlangefe** [comment]
We fixed this just yesterday on our dataset. We found the issue stemmed from how we were constructing the labels and annotation png files. We needed to make sure that the labels matched and each instance had a unique id. We also had to make sure they were encoded correctly by using the formula from COCO: id = R + G*256 + B*256^2. Once we fixed this, we actually get true positives in the confusion matrix, so we don't have issues like this anymore. Hope this is clear and helps. Feel free to ask more questions if this doesn't make sense. @RYYAI
